apps/ewallet_api/lib/ewallet_api/v1/controllers/transfer_controller.ex

```diff
@@ -19,11 +18,43 @@ defmodule EWalletAPI.V1.TransferController do
     attrs
     |> Map.put("idempotency_token", conn.assigns[:idempotency_token])
     |> TransactionGate.process_with_addresses()
-    |> respond_with_balances(conn)
+    |> respond_with(:balances, conn)
   end
 
   def transfer(conn, _attrs), do: handle_error(conn, :invalid_parameter)
 
+  def transfer_for_user(
+        conn,
+        %{
+          "to_address" => to_address,
+          "token_id" => token_id,
+          "amount" => amount
+        } = attrs
+      )
+      when to_address != nil and token_id != nil and is_integer(amount) do
+    balance_res = BalanceFetcher.get(conn.assigns.user, attrs["from_address"])
+
+    case balance_res do
```

Comment:
If we go with this, I don't want to see those functions in this file - I believe controller shouldn't be spammed with this kind of pattern matching functions because it slowly turns them into unreadable monsters. 

As long as it's in a case, it's still pretty easy to reason with, but if we add four more functions, let's add them in a new/existing module where they belong.

Reply:
So from additional chatting...

Arguments for `case`:
- The controller action is complete within itself and there's no need to look anywhere else to understand what the action is doing. Splitting to smaller functions can be harder to read as it's hard to see the dependencies between functions.
- The cases here are simple enough with only the successful case that has significant logic.

Arguments for pattern matching functions:
- Avoiding large chunks of code and splitting them to really short functions help maintain complexity of each function (less moving variables, less logic).

Conclusion:
- For this PR, up to @mederic-p 
- We will experiment with one controller action per file and encourage pattern matching private functions. This should help preventing the confusion from having lots of functions, as well as improving the overall code readability/maintainability.